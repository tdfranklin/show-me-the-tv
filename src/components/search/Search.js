import React, { Component } from 'react';
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import TVShowResults from '../tvShowResults/TVShowResults';
import axios from 'axios';

class Search extends Component {
    constructor(props) {
        super(props);
        this.state = {
            searchText: '',
            shows: [],
            popular: []
        };
    };

    componentDidMount() {
        axios
            .get('/api/popular')
            .then(res => this.setState({shows: res.data, popular: res.data}))
            .catch(err => console.log(err));
    }

    onTextChange = e => {
        const value = e.target.value;
        this.setState({ [e.target.name]: value }, () => {
            if (value === '') {
            this.setState({ shows: this.state.popular });
            } else {
                axios
                    .get(`/api/search/${value}`)
                    .then(res => this.setState({ shows: res.data }))
                    .catch(err => console.log(err));
            } 
        });
    }

    render() {
        return (
            <div>
                <TextField
                    name='searchText'
                    value={this.state.searchText}
                    onChange={this.onTextChange}
                    floatingLabelText="Search for TV Shows"
                    fullWidth={true}
                />
                <br />
                {this.state.shows.length > 0 ? (<TVShowResults shows={this.state.shows} />) : null }
            </div>
        );
    }
}

export default Search;