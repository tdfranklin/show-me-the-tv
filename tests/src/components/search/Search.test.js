import React from 'react';
import ReactDOM from 'react-dom';
import Search from '../../../../src/components/search/Search';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import TextField from 'material-ui/TextField';
import SelectField from 'material-ui/SelectField';
import { Tabs, Tab } from 'material-ui/Tabs';

configure({ adapter: new Adapter() });

describe('Search', () => {
    it('matches the snapshot', () => {
        const tree = shallow(<Search />);
        expect(toJson(tree)).toMatchSnapshot();
    });

    it('calls componentDidMount', () => {
        jest.spyOn(Search.prototype, 'componentDidMount');
        const wrapper = shallow(<Search />);
        expect(Search.prototype.componentDidMount.mock.calls.length).toBe(1);
    });

    describe('Material UI Components', () => {
        const wrapper = shallow(<Search />);

        describe('TextField', () => {
            let node = wrapper.find(TextField);
            
            it('renders component', () => {
                expect(node).toBeTruthy();;
            });

            it('changes label text based on content type', () => {
                const props = node.props();
                expect(props.floatingLabelText).toBe("Search for TV Shows");
                wrapper.setState({value: 'movies'});
                node = wrapper.find(TextField);
                expect(node.props().floatingLabelText).toBe("Search for Movies");
            });

            it('updates value based on state change', () => {
                expect(node.props().value).toBe('');
                wrapper.setState({searchText: 'House'});
                node = wrapper.find(TextField);
                expect(node.props().value).toBe('House');
            });
        });

        describe('Tabs', () => {
            const node = wrapper.find(Tabs);
            
            it('renders component', () => {
                expect(node).toBeTruthy();;
            });
        });

        describe('Tab', () => {
            const node = wrapper.find(Tab);
            
            it('renders component', () => {
                expect(node).toBeTruthy();
            });
        });
    });
});