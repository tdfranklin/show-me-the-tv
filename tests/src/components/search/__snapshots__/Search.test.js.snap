// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches the snapshot 1`] = `
<div>
  <TextField
    disabled={false}
    floatingLabelFixed={false}
    floatingLabelText="Search for TV Shows"
    fullWidth={true}
    multiLine={false}
    name="searchText"
    onChange={[Function]}
    rows={1}
    type="text"
    underlineShow={true}
    value=""
  />
  <br />
  <Tabs
    initialSelectedIndex={0}
    onChange={[Function]}
    style={
      Object {
        "marginBottom": "1%",
      }
    }
    value="tv"
  >
    <Tab
      label="TV"
      style={
        Object {
          "background": "#1c1d1d",
          "color": "darkgrey",
          "fontSize": "20px",
        }
      }
      value="tv"
    />
    &gt;
    <Tab
      label="Movies"
      style={
        Object {
          "background": "#1c1d1d",
          "color": "darkgrey",
          "fontSize": "20px",
        }
      }
      value="movies"
    />
    &gt;
  </Tabs>
</div>
`;
